pkg/signature/kms/cliplugin/README.md

```diff
@@ -0,0 +1,100 @@
+# CLI Plugin
+
+This is a package and module for using custom KMS plugins as separate executables.
+It is intended to be used by cosign, but you may use this in your own programs that import sigstore.
+
+## Design
+
+We follow [kubectl's style](https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/#writing-kubectl-plugins) of plugins. Any language that can create CLI programs can be used to make a plugin.
+
+### Usage
+
+Plugins are separate programs on your system's PATH, named in the scheme `sigstore-kms-[name]`, like `sigstore-kms-my-hsm`. They can be invoked with cosign like `cosign [sub-command] --key "my-hsm://my-key-id" ...
+`
+
+### Protocol
+
+The main program will invoke the program with these specifications:
+
+* stdin
+  * Data to be signed or verified.
+* arg 1
+  * A number idnetifying the version of this protocol.
+    * In the future we may change this protocol, either the encoding or the formatting of argument and return values.
+    These are necessarily breaking changes, and will be major-version bumps to sigstore and cosign.
```

Comment:
```suggestion
    Protocol changes will result in a major-version bump for the library.
```

Reply:
Removing a mention of Cosign, as the maintainers may not want to do a major bump for this.

Reply:
fixed.